tinygrad/renderer/cstyle.py

```diff
@@ -167,7 +167,7 @@ def ssa(prefix:str, u:Optional[UOp]=None):
             val = self.render_cast(precast, dtype, bitcast=True)
           elif uop is UOps.CAST: val = self.render_cast(r[src[0]], dtype, bitcast=False)
           else: val = self.render_vectorize([r[x] for x in src], dtype)
-          if child_count[u] <= 1: r[u] = val
+          if child_count[u] - len([x.op is UOps.IF and len(x.src) > 1 and u in x.src[1:] for x in u.parents]) <= 1: r[u] = val
```

Comment:
This was needed to stop unnecessary variables from being created (sometimes out of scope/in a nested loop) when it was only referenced by the STORE inside the code, and by the IF only for correct graph structure/ordering.

Reply:
OK, can you move this to a separate diff with a test?
Your diffs should have the essential lines to get **correctness**. Net noop changes are always separate PRs.

Reply:
is this the thing that's doing
![image](https://github.com/user-attachments/assets/ece78d58-b901-4260-9b85-f9c948e3a069)
?

Reply:
> OK, can you move this to a separate diff with a test?

Sounds good, I'll open up a PR specifically for this today.

> is this the thing that's doing

Ya that change would be due to this line modification.

The only reason I needed to include it is due to `test_padto_where_multireduce` failing with my new additions. It feels like a more subtle preexisting  bug that I might've ran into where a STORE src's could be out of scope if another UOp also references it, causing it to become a variable.

This patch only addresses it for the case of when an IF UOp refers to it, but I think there's the possibility of it bubbling up elsewhere.